rDNA/tests/testthat/test-data-management.R

```diff
@@ -0,0 +1,75 @@
+context("Data management")
+
+conn <- dna_connection("sample.dna")
+
+# Not working
```

Comment:
Why is this not working? Do I need to fix this?

Reply:
Basically `capture.output()` is not working with `dna_addAttribute` and `dna_addDocument` yet, which means the output can't be compared to a string. I assumed you would apply the fix with `JRI.jar` to this as well at some point. Then the test can be activated.

Reply:
Weird, it should work with all functions in Java. I did not fix it for any specific function but divert the output by setting up the JRI output stream once in the beginning when `dna_init` is called. I need to test this again at some point.

Reply:
No, I don't think that helped. I pulled the changes and compiled a new `.jar` but still doesn't work:

```
library("rDNA")
dna_init("dna-2.0-beta23.jar")
conn <- dna_connection(dna_sample(overwrite = T))
x <- capture.output(dna_addAttribute(conn))
x
```

Reply:
I just checked, and it seems this is not a problem on the Java side of things. If you look at the `dna_addAttribute` function in rDNA, the function produces a status message about the attribute having been created. I don't know why `capture.output` doesn't capture this message. Weird.

But it's still good that I fixed the `System.err` output stream for other purposes...

Reply:
I see. The problem is that you use `message` instead of `cat` in `dna_addAttribute`. Is there a specific reason for this?

Reply:
In this, example, the output is not captured either:
```
mytest <- function() {
  message("hello")
}
x <- capture.output(mytest())
x
```
However, it works if you add a `type` argument:
```
mytest <- function() {
  message("hello")
}
x <- capture.output(mytest(), type = "message")
x
```
This is weird because the default of the argument is already `type = c("output", "message")`. But whatever, just use that argument...

Reply:
The general advice in the CRAN R package guidelines is to use message for generic output while cat is reserved only for show and print methods, as far as I can remember.

Reply:
I don't find this in the CRAN docs, but see chapter 3.3.3 of this book: https://csgillespie.github.io/efficientR/programming.html

Reply:
Ah, good to know! There is a specific test for messages, so it's not a problem to fix this.